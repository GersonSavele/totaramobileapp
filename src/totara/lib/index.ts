/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import config from "./config";
import { Logger, Log } from "./logger";
import { showMessage } from "./tools";
import * as constants from "./constants";

export { config, Logger, Log, showMessage, constants };
