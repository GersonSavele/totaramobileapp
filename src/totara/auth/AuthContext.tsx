/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Jun Yamog <jun.yamog@totaralearning.com
 *
 */


import React from "react";

import { config } from "@totara/lib";
import { Theme } from "@totara/theme";

export const AuthContext = React.createContext<State>({
  isAuthenticated: false,
  appState: undefined,
  logOut: () => {
    return Promise.resolve();
  }
});

export type State = {
  isAuthenticated: boolean;
  appState?: AppState;
  logOut: () => Promise<void>;
};

export type SiteInfo = {
  auth: string;
  siteMaintenance: boolean;
  theme?: Theme;
  version: string;
};

export type AppState = {
  apiKey: string;
  host: string;
  siteInfo: SiteInfo;
};

enum Compatible {
  Api = 1
}

//TODO-Need to integrate correct logic
export const isValidApiVersion = (apiVersoin?: string) => {
  const compatibilityList = isCompatible(apiVersoin);
  return compatibilityList.length > 0;
};

export const isCompatible = (version?: string) => {
  const fullCompatible = [Compatible.Api];
  if (config.minApiVersion === "disabled") {
    return fullCompatible;
  } else {
    if (version && config.minApiVersion <= version) 
      return [Compatible.Api];
    else 
    return [];
  }
};