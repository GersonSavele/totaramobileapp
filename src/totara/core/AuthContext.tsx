/*
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Jun Yamog <jun.yamog@totaralearning.com>
 */


import React from "react";

import { config } from "@totara/lib";

import { AuthContextState, initialState, Setup } from "./AuthHook";

export const AuthContext = React.createContext<State>({
  authContextState: initialState,
  logOut: () => {
    return Promise.resolve();
  },
  onLoginSuccess: () => {},
  onLoginFailure: () => {},
});

type State = {
  authContextState: AuthContextState;
  logOut: (local?: boolean) => Promise<void>;
  onLoginSuccess: (setup: Setup) => void;
  onLoginFailure: (error: Error) => void;
}

enum Compatible {
  Api = 1
}

export const AuthConsumer = AuthContext.Consumer;

//TODO MOB-273 Need to integrate correct logic
export const isValidApiVersion = (apiVersoin?: string) => {
  const compatibilityList = isCompatible(apiVersoin);
  return compatibilityList.length > 0;
};

export const isCompatible = (version?: string) => {
  const fullCompatible = [Compatible.Api];
  if (config.minApiVersion === "disabled") {
    return fullCompatible;
  } else {
    if (version && config.minApiVersion <= version) 
      return [Compatible.Api];
    else 
    return [];
  }
};