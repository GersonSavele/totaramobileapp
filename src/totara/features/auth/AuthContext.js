/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author Jun Yamog <jun.yamog@totaralearning.com
 *
 */

import React from "react";
import {AsyncStorage, Text} from "react-native";
import {ApolloClient}  from "apollo-client";
import {ApolloProvider} from "react-apollo";
import {ApolloLink} from 'apollo-link';
import {RetryLink} from 'apollo-link-retry';
import {HttpLink} from 'apollo-link-http';
import {InMemoryCache} from "apollo-cache-inmemory";
import SplashScreen from "react-native-splash-screen";
import PropTypes from 'prop-types';

import {config} from "@totara/lib";

import AuthLogin from "./AuthLogin";


const AuthContext = React.createContext(
  {
    setWebSession: () => {},
    webSession: undefined,
    setApiKey: () => {},
    apiKey: undefined,
    logOut: () => {}
  }
);

const AuthConsumer = AuthContext.Consumer;

/**
 * AuthProvider will provide the auth states and data needed for any components inside
 * the scope.  Auth states, methods should be use with AuthConsumer
 *
 * This will also provide the switching from a component used to authenticate a user,
 * Once authenticated it would load the children with an ApolloProvider with a authenticated
 * client.
 *
 * @example
 *
 * <AuthProvider>
 *   <ElementsThatWillHaveAuthenticatedUserAndGraphQLApi/>
 * </AuthProvider>
 *
 */
class AuthProvider extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      setWebSession: this.setWebSession,
      webSession: undefined,
      setApiKey: this.setApiKey,
      apiKey: undefined,
      logOut: this.logOut,
      isLoading: true
    };

    this.bootstrap();
  }

  bootstrap = async () => {
    const apiKey = await AsyncStorage.getItem('apiKey');

    SplashScreen.hide();

    this.setState({
      apiKey: apiKey,
      isLoading: false
    });
    console.log("state", this.state);
    // TODO add some logging and error handling, important routine
  };

  setWebSession = async (webSession) => {
    await this.getAndStoreApiKey("uegkuxhdeHaoc0dYX9HwswvisvUIEb");

    console.log("state", this.state);
  };

  logOut = async () => {
    await AsyncStorage.clear();
    this.setState({
      apiKey: undefined
    });
  };

  link = ApolloLink.from([
    new RetryLink({attempts: {max: 10}}),
    new HttpLink({ uri: config.mobileApi + "/graphql" })
  ]);

  client = new ApolloClient({
    link: this.link,
    cache: new InMemoryCache()
  });

  getAndStoreApiKey = async (setupSecret) => {
    try {
      const apiKey = await fetch("http://totara72/totara/mobile/device_register.php", {
        method: "POST",
        body: JSON.stringify({
          setupsecret: setupSecret
        })
      }).then((response) => {
        if (response.status === 200) return response.json();
        throw new Error(`Server Error: ${response.status}`);
      }).then((json) => json.data.apikey);

      await AsyncStorage.setItem("apiKey", apiKey);

      this.setState({
        apiKey: apiKey
      });

      return apiKey;
    } catch (error) {
      console.error(error);
      // TODO add some logging and error handling, important routine

      return undefined;
    }
  };

  render() {
    return(
      <AuthContext.Provider value={this.state}>
        {
          (this.state.isLoading)
            ? <Text>auth loading, this text should not be seen unless bootstrap failed</Text>
            : (this.state.apiKey)
              ? <ApolloProvider client={this.client}>
                  {this.props.children}
                </ApolloProvider>
              : <AuthLogin setWebSession={this.setWebSession}/>
        }
      </AuthContext.Provider>
    )
  }
}



export {AuthProvider, AuthConsumer}