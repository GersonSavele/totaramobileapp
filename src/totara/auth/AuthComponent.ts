/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React from "react";

import { Setup } from "@totara/core/AuthHook";

export class AuthComponent<P = {}, S = {}> extends React.Component<AuthFlowChildProps & P, S> {}

export type AuthFlowChildProps = {
  siteUrl?: string;
  onLoginSuccess: (setup: Setup) => Promise<void>;
  onLoginFailure: (error: Error) => Promise<void>;
};
