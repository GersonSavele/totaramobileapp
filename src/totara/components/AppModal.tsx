/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author: Kamala Tennakoon <kamala.tennakoon@totaralearning.com>
 */

import React, { useContext, useState } from "react";

import { translate } from "@totara/locale";
import { AuthContext, Setup } from "@totara/auth/AuthContext";
import { InfoModal} from "./infoModal"
import PrimaryButton from "./PrimaryButton";
import { config } from "@totara/lib";

enum Compatible {
  Api = 1
}

const AppModal = () => {
  const {setup} = useContext(AuthContext);
  const isShowIncompatibleApi = !isValidApiVersion(setup);
  const [isVisible, setIsVisible] = useState(isShowIncompatibleApi);
  if(isVisible) 
    return (
    <InfoModal title={translate("message.sorry")} description={translate("message.incompatible_api")} imageType={"url_not_valid"} visible={isVisible}>
      <PrimaryButton  text={translate("general.try_again")} onPress={() => { setIsVisible(!isVisible) }} />
    </InfoModal>
    );
  else 
    return null;
};

//TODO-Need to integrate correct logic
export const isValidApiVersion = (setup?: Setup) => {
  if (setup && setup.apiVersion) {
    const compatibilityList = isCompatible(setup.apiVersion)
    return compatibilityList.length > 0
  }
  return false;
};

export const isCompatible = (version: string) => {
  if (config.minApiVersion <= version)
    return [Compatible.Api]
  else 
    return []
};

export default AppModal;