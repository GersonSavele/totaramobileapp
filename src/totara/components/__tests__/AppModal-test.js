/**
 * This file is part of Totara Mobile
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @author: Kamala Tennakoon <kamala.tennakoon@totaralearning.com>
 */

import React from "react";
import renderer from "react-test-renderer";

import AppModal from "../AppModal";
import { AuthContext } from "@totara/auth/AuthContext";
import { config } from "@totara/lib";

describe("AppModal", () => {
  config.minApiVersion = "2019101802";
  it("screens for different apiVersion", async () => {
    const validSetupVersion = { apiVersion: "2019101802"};
    const validVersionComponent = renderer.create(
      <AuthContext.Provider value={{setup: validSetupVersion}} >
        <AppModal />
      </AuthContext.Provider>
    );
    expect(validVersionComponent.toJSON()).toMatchSnapshot();

    const oldSetupVersion = { apiVersion: "2010101802"};
    const oldVersionComponent = renderer.create(
      <AuthContext.Provider value={{setup: oldSetupVersion}} >
        <AppModal />
      </AuthContext.Provider>
    );
    expect(oldVersionComponent.toJSON()).toMatchSnapshot();

    const higherSetupVersion = { apiVersion: "2030101802"};
    const higherVersionComponent = renderer.create(
      <AuthContext.Provider value={{setup: higherSetupVersion}} >
        <AppModal />
      </AuthContext.Provider>
    );
    expect(higherVersionComponent.toJSON()).toMatchSnapshot();

    config.minApiVersion = "disabled";
    const disabledAppMinVersionComponent = renderer.create(
      <AuthContext.Provider value={{setup: validSetupVersion}} >
        <AppModal />
      </AuthContext.Provider>
    );
    expect(disabledAppMinVersionComponent.toJSON()).toMatchSnapshot();
  });
});
