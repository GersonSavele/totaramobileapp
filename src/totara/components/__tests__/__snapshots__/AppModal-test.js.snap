// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal screens for different apiVersion 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "flex": 1,
              "marginHorizontal": 20,
              "marginVertical": 32,
            },
            Object {
              "backgroundColor": "#FFFFFF",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginLeft": "5%",
              "marginRight": "5%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/images/general_error/general_error.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 375,
                    "resizeMode": "contain",
                    "width": 525,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 44,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.title with undefined
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 16,
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.description with undefined
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "justifyContent": "space-between",
                  "marginVertical": 16,
                  "minHeight": 104,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AppModal screens for different apiVersion 2`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "flex": 1,
              "marginHorizontal": 20,
              "marginVertical": 32,
            },
            Object {
              "backgroundColor": "#FFFFFF",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginLeft": "5%",
              "marginRight": "5%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/images/general_error/general_error.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 375,
                    "resizeMode": "contain",
                    "width": 525,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 44,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.title with undefined
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 16,
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.description with undefined
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "justifyContent": "space-between",
                  "marginVertical": 16,
                  "minHeight": 104,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AppModal screens for different apiVersion 3`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "flex": 1,
              "marginHorizontal": 20,
              "marginVertical": 32,
            },
            Object {
              "backgroundColor": "#FFFFFF",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginLeft": "5%",
              "marginRight": "5%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/images/general_error/general_error.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 375,
                    "resizeMode": "contain",
                    "width": 525,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 44,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.title with undefined
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 16,
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.description with undefined
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "justifyContent": "space-between",
                  "marginVertical": 16,
                  "minHeight": 104,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`AppModal screens for different apiVersion 4`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "flex": 1,
              "marginHorizontal": 20,
              "marginVertical": 32,
            },
            Object {
              "backgroundColor": "#FFFFFF",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginLeft": "5%",
              "marginRight": "5%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/resources/images/general_error/general_error.png",
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "height": 375,
                    "resizeMode": "contain",
                    "width": 525,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginVertical": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 44,
                        "fontWeight": "500",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.title with undefined
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "paddingBottom": 16,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#3D444B",
                        "fontSize": 16,
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  translated general_error_feedback-modal.description with undefined
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "justifyContent": "space-between",
                  "marginVertical": 16,
                  "minHeight": 104,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
              <View
                accessibilityRole="button"
                accessibilityStates={Array []}
                accessible={true}
                clickable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "padding": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    this is a mock button
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
